modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/BaseStructuredContentResourceImpl.java

```diff
@@ -592,6 +599,51 @@ public Rating putStructuredContentMyRating(
 		return new Rating();
 	}
 
+	/**
+	 * Invoke this method with the command line:
+	 *
+	 * curl -X 'PUT' 'http://localhost:8080/o/headless-delivery/v1.0/structured-contents/{structuredContentId}/permissions'  -u 'test@liferay.com:test'
+	 */
+	@Override
+	@PUT
+	@Parameters(
+		value = {
+			@Parameter(in = ParameterIn.PATH, name = "structuredContentId")
+		}
+	)
+	@Path("/structured-contents/{structuredContentId}/permissions")
+	@Produces({"application/json", "application/xml"})
+	@Tags(value = {@Tag(name = "StructuredContent")})
+	public void putStructuredContentPermission(
+			com.liferay.portal.vulcan.permissions.Permissions[] permissions,
+			@NotNull @Parameter(hidden = true) @PathParam("structuredContentId")
+				Long structuredContentId)
+		throws Exception {
+
+		PermissionChecker permissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		String resourceName = getStructuredContentResourceName();
+
+		if (!permissionChecker.hasPermission(
+				0, resourceName, 0, ActionKeys.PERMISSIONS)) {
+
+			return;
+		}
+
+		resourcePermissionLocalService.updateResourcePermissions(
+			contextCompany.getCompanyId(), 0, resourceName,
```

Comment:
You are passing groupId here as 0 too. Permissions are usually scoped by groupId, I guess, because we are passing the entity's groupId too.

Because of these differences, I will have to override this method entirely in *ResourceImpl.

Reply:
This is the groupId I was referring to, that is irrelevant:

```
	if (roleName.equals(RoleConstants.PLACEHOLDER_DEFAULT_GROUP_ROLE)) {
			if (groupId == 0) {
				throw new NoSuchRoleException(
					"Specify a group ID other than 0 for role name " +
						RoleConstants.PLACEHOLDER_DEFAULT_GROUP_ROLE);
			}

			return roleLocalService.getDefaultGroupRole(groupId);
		}
```

Reply:
It's only used if you pass PLACEHOLDER_DEFAULT_GROUP_ROLE as the role name